src/platform_impl/web/event_loop/runner.rs

```diff
@@ -108,7 +113,22 @@ impl<T: 'static> Shared<T> {
             destroy_pending: RefCell::new(VecDeque::new()),
             scale_change_detector: RefCell::new(None),
             unload_event_handle: RefCell::new(None),
-        }))
+            proxy_sender,
+        }));
+
+        wasm_bindgen_futures::spawn_local({
+            let runner = this.clone();
+            async move {
+                while let Ok(value) = (&mut proxy_receiver).await {
```

Comment:
I think it would be more idiomatic to have an `async fn next()` on the receiver and to poll that instead. Futures aren't really meant to be polled after they're completed.

Reply:
I needed some way to yield, so instead of writing a custom [`yield_now`](https://docs.rs/tokio/latest/tokio/task/fn.yield_now.html) I just used [`poll_fn`](https://doc.rust-lang.org/stable/core/future/fn.poll_fn.html), which was stabilized in v1.64, which we just bumped to just recently anyway https://github.com/rust-windowing/winit/pull/2723.